File: validation/src/test/groovy/io/micronaut/docs/datavalidation/pogo/Email.java
package io.micronaut.docs.datavalidation.pogo;

import javax.validation.constraints.NotBlank;

public class Email {

    @NotBlank // <1>
    String subject;

    @NotBlank // <1>
    String recipient;

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getRecipient() {
        return recipient;
    }

    public void setRecipient(String recipient) {
        this.recipient = recipient;
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/datavalidation/pogo/Email.kt
package io.micronaut.docs.datavalidation.pogo

import io.micronaut.core.annotation.Introspected

import javax.validation.constraints.NotBlank

@Introspected
open class Email {

    @NotBlank // <1>
    var subject: String? = null

    @NotBlank // <1>
    var recipient: String? = null
}
